frontend/src/metabase/components/form/widgets/FormDateWidget/FormDateWidget.tsx

```diff
@@ -62,7 +62,7 @@ const FormDateWidget = forwardRef(function FormDateWidget(
   return (
     <DateWidget
       ref={ref}
-      value={value}
+      value={value as Moment | undefined}
```

Comment:
Why do we need casting? I believe the definition should infer the correct type, so we don't cast it here

Reply:
@alxnddr Seems like it's a "moment" vs "moment-timezone" lib thing.

It does complain if we don't cast.

If we edit the `DateWidget` file to use "moment-timezone", and conform to "lib/time", it solves this type flag.
But it opens a cascade of wrong types down the line, probably because of the same thing.

For now I went with this compromise so we can make other files import "moment-timezone" as well, instead of "moment".
This has a chance of introducing regressions, so better do it separately.